static/absolute/js/searchPage.js

```diff
@@ -110,45 +110,54 @@ socket.on("refresh", () => {
 
 async function refreshUI() {
   const eventId = getEventId();
-  const participantsResponse = await (
-    await fetch(`/api/${eventId}/participants`)
-  ).json();
-  const restaurantsResponse = await (
-    await fetch(`/api/${eventId}/restaurants`)
-  ).json();
 
-  initMap(participantsResponse, restaurantsResponse);
-  showRestaurants(restaurantsResponse);
+  // Fetch in parallel
+  const [participantsResponse, restaurantsResponse] = await Promise.all([
+    fetch(`/api/${eventId}/participants`).then((response) => response.json()),
+    fetch(`/api/${eventId}/restaurants`).then((response) => response.json()),
+  ]);
+
+  if (
+    participantsResponse.status !== 200 ||
+    restaurantsResponse.status !== 200
+  ) {
+    console.error(`Failed to fetch from endpoints`);
+    return;
+  }
+
+  const { data: participants } = participantsResponse;
+  const {
+    data: { places, center },
+  } = restaurantsResponse;
+
+  const domNodes = showRestaurants(places);
+  initMap(participants, center, places, domNodes);
 }
 
 /**
  * Creates HTML elements for the restaurant details and adds it
  * to a container in searchResults.html.
- *
- * For now, this function deals with hard-coded data, but this
- * can be used as a template for when we get data the Places Library results.
  */
-function showRestaurants(restaurantsResponse) {
-  const allRestaurants = restaurantsResponse.data;
+function showRestaurants(restaurants) {
   const restaurantContainer = document.getElementById("restaurant-container");
 
   restaurantContainer.innerHTML = "";
 
-  if (!allRestaurants.hasOwnProperty("results")) {
+  if (restaurants.length === 0) {
```

Comment:
Is there a way to distinguish the difference between having no restaurants when no one has put in information and when no results were found? I think the instructions message would be confusing otherwise.

Reply:
Nevermind. I believe the error result is still getting returned when the response is not "OK".